File: bindings/com/intel/rs/Distortion.java
/*
    INTEL CORPORATION PROPRIETARY INFORMATION This software is supplied under the
    terms of a license agreement or nondisclosure agreement with Intel Corporation
    and may not be copied or disclosed except in accordance with the terms of that
    agreement.
    Copyright(c) 2015 Intel Corporation. All Rights Reserved.
*/

package com.intel.rs;

public enum Distortion
{
    NONE(0), // Rectilinear images, no distortion compensation required
    MODIFIED_BROWN_CONRADY(1), // Equivalent to Brown-Conrady distortion, except that tangential distortion is applied to radially distorted points
    INVERSE_BROWN_CONRADY(2); // Equivalent to Brown-Conrady distortion, except undistorts image instead of distorting it
    public final int code;
    private Distortion(int code) { this.code = code; }
}

File: wrappers/csharp/Intel.RealSense/Types/Enums/Distortion.cs
// License: Apache 2.0. See LICENSE file in root directory.
// Copyright(c) 2017 Intel Corporation. All Rights Reserved.

namespace Intel.RealSense
{
    /// <summary>
    /// Distortion model: defines how pixel coordinates should be mapped to sensor coordinates.
    /// </summary>
    public enum Distortion
    {
        /// <summary> Rectilinear images. No distortion compensation required.</summary>
        None = 0,

        /// <summary> Equivalent to Brown-Conrady distortion, except that tangential distortion is applied to radially distorted points</summary>
        ModifiedBrownConrady = 1,

        /// <summary> Equivalent to Brown-Conrady distortion, except undistorts image instead of distorting it</summary>
        InverseBrownConrady = 2,

        /// <summary> F-Theta fish-eye distortion model</summary>
        Ftheta = 3,

        /// <summary> Unmodified Brown-Conrady distortion model</summary>
        BrownConrady = 4,

        /// <summary Four parameter Kannala Brandt distortion model</summary>
        KannalaBrandt4 = 5,

    }
}